interfaces/hotplug/deviceinfo.go

```diff
@@ -26,10 +26,14 @@ import (
 	"github.com/snapcore/snapd/dirs"
 )
 
+type hotplugDeviceInfoData struct {
```

Comment:
any deep reason to introduce this new type?

Reply:
The only reason is that (AFAIU) json marshalling needs attributes to be exported in order to serialize/deserialize them. By having an embedded structure like this I can avoid exposing data. Is there any alternative that I missed?

Reply:
I see, thanks